### ISSUE TITLE
not working with sbt dependencies

### ISSUE THREAD

**Issue by fdietze** [megaannum/vimside#27]
Hi,

I'm trying to get vimside working with my current project. Here is my setup:
sbt 0.12.4
scala 2.10.3
ensime sbt cmd 0.1.1
ensime 0.9.8.9
vimside installed with vundle

It is working fine with a project relying only on the standard library. But when I'm adding dependencies in my build.sbt, they are not recognized.

It works with Sublime text and ensime on the same project.

Thanks for your help
Felix

**Comment by megaannum** [comment]
I am far from a sbt expert. For Vimside, I simply
copied the functionality that ENSIME provided as
best that I could. I have never used dependences with
sbt with or without Vimside, so I do not know
where to start providing help.

I suspect that if you asked the ENSIME list you
would not get a response since they would assume
(maybe rightly) that it was the Vimside implementation
that was at fault.

It is possible that either or both of the vimside and
ensime logs (VIMSIDE_LOG and ENSIME_LOG) might
indicate what part of the sbt file was being read and
forwarded from the Vim code (Vimside) to the backend
Java engine (Ensime).

Richard

On 12/13/2013 10:56 AM, Felix Dietze wrote:

> Hi,
> 
> I'm trying to get vimside working with my current project. Here is my setup:
> sbt 0.12.4
> scala 2.10.3
> ensime sbt cmd 0.1.1
> ensime 0.9.8.9
> vimside installed with vundle
> 
> It is working fine with a project relying only on the standard library.
> But when I'm adding dependencies in my build.sbt, they are not recognized.
> 
> It works with Sublime text and ensime on the same project.
> 
> Thanks for your help
> Felix
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/megaannum/vimside/issues/27.

## 

Quis custodiet ipsos custodes

**Comment by fdietze** [comment]
I don't think it has to do with sbt, this is just how I got there. With an sbt plugin I generated this .ensime file:

```
(
  :subprojects (
    (
      :name "test"
      :module-name "test"
      :depends-on-modules (

      )
      :package "test"
      :version "0.1-SNAPSHOT"
      :compile-deps (
        "/home/felix/.sbt/boot/scala-2.10.3/lib/scala-library.jar"
        "/home/felix/.ivy2/cache/com.typesafe.akka/akka-actor_2.10/jars/akka-actor_2.10-2.2.3.jar"
        "/home/felix/.ivy2/cache/com.typesafe/config/bundles/config-1.0.2.jar"
      )
      :runtime-deps (
        "/home/felix/.sbt/boot/scala-2.10.3/lib/scala-library.jar"
        "/home/felix/.ivy2/cache/com.typesafe.akka/akka-actor_2.10/jars/akka-actor_2.10-2.2.3.jar"
        "/home/felix/.ivy2/cache/com.typesafe/config/bundles/config-1.0.2.jar"
        "/home/felix/test/target/scala-2.10/classes"
        "/home/felix/test/target/scala-2.10/classes"
      )
      :test-deps (
        "/home/felix/.sbt/boot/scala-2.10.3/lib/scala-library.jar"
        "/home/felix/.ivy2/cache/com.typesafe.akka/akka-actor_2.10/jars/akka-actor_2.10-2.2.3.jar"
        "/home/felix/.ivy2/cache/com.typesafe/config/bundles/config-1.0.2.jar"
        "/home/felix/test/target/scala-2.10/classes"
        "/home/felix/test/target/scala-2.10/test-classes"
      )
      :source-roots (
        "/home/felix/test/src/main/scala"
        "/home/felix/test/src/main/java"
        "/home/felix/test/target/scala-2.10/src_managed/main"
        "/home/felix/test/src/test/scala"
        "/home/felix/test/src/test/java"
        "/home/felix/test/target/scala-2.10/src_managed/test"
      )
      :target "/home/felix/test/target/scala-2.10/classes"
      :test-target "/home/felix/test/target/scala-2.10/test-classes"
    )
  )
)
```

With this file I have auto completion for Akka in Sublime Text with ensime, but not in vimside. In vimside I only get auto completion for the scala standard library. I hope this is clear now.

**Comment by fdietze** [comment]
Any Idea for this one?

**Comment by megaannum** [comment]
I have not had any time to work on it.
It had been my hope/expectation that as Vimside developed
others would contribute - but that has not happened.

On 01/02/2014 11:00 AM, Felix Dietze wrote:

> Any Idea for this one?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/megaannum/vimside/issues/27#issuecomment-31475276.

## 

Quis custodiet ipsos custodes
